### PR TITLE
Feature/custatevec

### PR DESCRIPTION
**Pull request by brian-dellabetta**

### CODE REVIEW

**Review by max-radin — COMMENTED** [review]
Hey brian this looks great! I do have one caveat about the comparison to cirq's built-in simulation. Under the hood, qe-cirq's `get_wavefunction` method [uses](https://github.com/zapatacomputing/qe-cirq/blob/34a5d215528a6a1e214c69141ac421f950455943/src/python/qecirq/simulator/simulator.py#L194-L196) the `final_state_vector` method of cirq's circuit class, which is different from cirq's official simulator, `cirq.Simulator`. My suspicion is that they perform similarly, but I don't know for sure. If you wanted to check this, you could just directly run the circuit using `cirq.Simulator.simuate` as you already do with the qsim simulator.

**Comment by brian-dellabetta** [comment]
Thanks @max-radin for the tip. I will try that as well to confirm it doesn't change much

**Review by mkodrycka — APPROVED** [review]
LGTM!

File: custatevec/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Mounted gpu has cuda v11.6
+FROM nvcr.io/nvidia/cuquantum-appliance:22.02-cirq
+
+WORKDIR /app
+
+# https://askubuntu.com/questions/1408016/the-following-signatures-couldnt-be-verified-because-the-public-key-is-not-avai
+RUN apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub && \
+    apt-get update
+
+# Install python/pip and git/cmake (needed for qsim-custatevec build)
```
Comment (mkodrycka) [inline custatevec/Dockerfile:10]: Do we need commented lines?

**Pull request merged**